src/nvim/buffer.c

```diff
@@ -1737,7 +1739,6 @@ buf_T *buflist_new(char_u *ffname_arg, char_u *sfname_arg, linenr_T lnum, int fl
     buf->b_sfname = vim_strsave(sfname);
   }
 
-  clear_wininfo(buf);
```

Comment:
This only needs to be done in the case of reusing an existing buffer, and we're now doing it in the free_buffer_stuff call above.